pkg/controller/node/nodecontroller.go

```diff
@@ -1078,7 +1082,11 @@ func (nc *NodeController) cancelPodEviction(node *v1.Node) bool {
 func (nc *NodeController) evictPods(node *v1.Node) bool {
 	nc.evictorLock.Lock()
 	defer nc.evictorLock.Unlock()
-	return nc.zonePodEvictor[utilnode.GetZoneKey(node)].Add(node.Name, string(node.UID))
+	evictor := nc.zonePodEvictor[utilnode.GetZoneKey(node)]
+	if evictor == nil {
+		return true
```

Comment:
If the zone does not exist, there is nothing to evict. So the pods are evicted. So the `true` is correct. Otherwise, the `false` means the node is already queued for eviction. But it is not as there is no such zone.

Plus, repetitive invocation always returns true (as long as the zone does not exist). Which means as long as the zone does not exist, pods of the node in the zone are already evicted.